Complete hands on with Composer and Namespaces

nickj10 opened the pull request:
Use a custom docker image with the PHP image in the local environment. Prepare hands on exercise to learn more about composer and namespaces.

nickj10 committed:
feat: add local-environment

nickj10 committed:
feat: update the local environment

nickj10 committed:
feat: import User incorrectly

nickj10 committed:
more details about namespaces

nickj10 committed:
feat: use faker package

vcaballero-salle commented on README.md:
```diff
@@ -101,5 +101,252 @@ With this, we are defining a mapping from namespaces to directories. The `src/`
+<body>
+    <h1>Library system</h1>
+    <?php
+    $user = new User(1, $faker->name());
+    $book = new Book("Design Systems", "Design", "Alla Kholmatova", 2017);
```
I woud move that to the top. There is no pint on having this business logic here. Even if it is just instantiating the classes.

vcaballero-salle commented on Dockerfile:
```diff
@@ -0,0 +1,10 @@
+FROM php:8.0-fpm-alpine
```
You can delete this file

vcaballero-salle commented on Makefile:
```diff
@@ -0,0 +1,7 @@
+.PHONY: run
```
You can delete this file

vcaballero-salle commented on README_local-environment.md:
```diff
@@ -0,0 +1,114 @@
+# Local Environment
```
You can delete this file

vcaballero-salle commented on README.md:
````diff
@@ -101,5 +101,252 @@ With this, we are defining a mapping from namespaces to directories. The `src/`
+?>
+```
+
+You can observe that the first line is `declare(strict_types=1);`. By setting `strict_types=1`, you tell the engine that `int $x` means `$x must only be an int proper, no type coercion allowed.` You have the great assurance you're getting exactly and only what was given, without any conversion or potential loss.
````
"proper"? -> "property"?

nickj10 replied:
yes, it's a type. fixed!

vcaballero-salle commented on README.md:
```diff
@@ -101,5 +101,252 @@ With this, we are defining a mapping from namespaces to directories. The `src/`
+        return $this->id;
+    }
+
+    public function setId(int $id): self
+    {
+        $this->id = $id;
+        return $this;
+    }
```
If you don't need that, delete it.

vcaballero-salle commented on README.md:
````diff
@@ -101,5 +101,252 @@ With this, we are defining a mapping from namespaces to directories. The `src/`
+}
+```
+
+To be able to use both classes, you can either import them individually:
````
Remove "either"

vcaballero-salle commented on README.md:
````diff
@@ -101,5 +101,252 @@ With this, we are defining a mapping from namespaces to directories. The `src/`
+echo $book->getName();
+```
+
+There is a way to simplify this. Both of these classes are in the same namespace. You have previously defined in the `composer.json` the namespaces. Instead of having to use `require` for each file individually, you can take advantage of the namespaces. First, remove the following lines:
````
But, there is a way to simplify this...

vcaballero-salle commented on README.md:
````diff
@@ -101,5 +101,252 @@ With this, we are defining a mapping from namespaces to directories. The `src/`
+
+First, we want to generate fake data with the [**Faker**](https://github.com/fzaninotto/Faker/blob/master/readme.md) package. According to the documentation, you can start using the package by creating the Faker:
+```
+use Faker\Factory; // using the Factory class included in the Faker package
````
Explain why this thisrd-party package is also "found" thanks to the autoload.php

vcaballero-salle commented on docker-compose.yaml:
```diff
@@ -0,0 +1,45 @@
+      - "9030:9001"
+    volumes:
+      - .:/app
+    depends_on:
+      - db
+
+  db:
```
If you don't use the service, remove it

vcaballero-salle commented on docker-compose.yaml:
```diff
@@ -0,0 +1,45 @@
+    env_file:
+      - .env
+
+  admin:
```
Rev¡move service

vcaballero-salle commented on etc/php/php.ini:
```diff
@@ -0,0 +1,4 @@
+date.timezone = "UTC"
```
You are not using this configuration file because all configuration is in the image. Delete it.

vcaballero-salle commented on public/index.php:
```diff
@@ -0,0 +1,59 @@
+<body>
+    <h1>Library system</h1>
+    <?php
+    $user = new User(1, $faker->name());
+    $book = new Book("Design Systems", "Design", "Alla Kholmatova", 2017);
```
Move that to the top

vcaballero-salle commented on src/Model/User.php:
```diff
@@ -0,0 +1,35 @@
+        return $this->id;
+    }
+
+    public function setId(int $id): self
+    {
+        $this->id = $id;
+        return $this;
+    }
```
If you don't need this, remove

vcaballero-salle reviewed (changes requested):
Also, remove the Faker library because it is abandoned. You can use https://github.com/FakerPHP/Faker instead.

nickj10 committed:
chore: remove unnecessary files and services

nickj10 committed:
fixup! chore: remove unnecessary files and services

nickj10 committed:
fixup! fixup! chore: remove unnecessary files and services

nickj10 commented:
> Also, remove the Faker library because it is abandoned. You can use https://github.com/FakerPHP/Faker instead.

I've removed all unnecessary code, files and packages. I've changed it to this newer version of Faker and also, I've removed all unnecessary services. The application is still working as expected after the changes.

Also for Issue #2 , I'm already using `vcaballerosalle/pwii-php-fpm` image. Does the server have to be from `vcaballerosalle`? Or using `nginx:1.19.6-alpine` is enough?

vcaballero-salle committed:
chore: fix typo in README

vcaballero-salle reviewed (approved):
Indeed the server has to be form vcaballerosalle, all images have too. However, these are not created yet. I've assigned the issue to myself. This PR is ready to be merged.
